### DISCUSSION TITLE
How to think about structuring notes?

### DISCUSSION THREAD

**Discussion by Bullish-Design** [ahyatt/ekg#103] [Q&A]
I'm having a hard time "thinking" about how to use ekg. It seems very useful for quick jotting of notes, but I have a desire to make a flexibly structured "tree" of notes, starting from a root node and grabbing and organizing everything that's related and displaying it in one buffer. 

I'm able to create a root node and link related nodes, but I'm hoping for the ability to group/sort/order nodes underneath that root. Right now I'm stuck with the root node and a ton of random notes without any structure. 

Thoughts? Am I misinterpreting what this package is able to do? Or am I just missing something here?

**Comment by garyo** [comment]
My opinion as a new user only, so take it for what it's worth: I think the concept of ekg is more like zettelkasten (i.e. a knowledge graph, linked by tags) than a tree. It takes some inspiration from logseq, which is aggressively anti-tree ;-) 
They do have a point: knowledge graphs look different from any angle, so imposing one tree (year / class / class-session) will prohibit a different tree (topic / subtopic or book / chapter / quote). Just cross-linking everything with tags helps you find relations of all types.

That said, ekg is still young. Search is questionable, sorting is mostly nonexistent, tag hierarchy is only somewhat there, and there is no grouping as such (just tag all related notes with the group tag though).

**Comment by ahyatt** [comment]
Agreed with @garyo.  We just don't have a lot of functionality with hierarchy in mind.  We do have tag hierarchies, but no navigation between them.  I personally don't think strict hierarchical ontologies are a good idea, which is why I haven't been focused on this.

**Reply by Bullish-Design** [reply]
Quite alright! I appreciate the response and the time saved.

@ahyatt - Do you think your triples database package could be well suited for generating/interacting with a hierarchical note structure like this?

I really want to explore this "flexibly hierarchical" concept, I'm just wondering if it would be more worthwhile forking ekg or starting off on my own closer to the database - and if so, how low to go.

Any thoughts or learnings you could lend from your experience building ekg and triples would be much appreciated.

**Reply by ahyatt** [reply]
Forking off ekg might be a good way to start.  The triples package would be a good backend - since it's a graph database, it can represent trees very easily.  The difficult issues in my opinion are design and creating reasonable UIs.  Emacs doesn't have good libraries to build sophisticated UIs, unfortunately.  Hopefully if you fork ekg you can re-use things we already built.  On the other hand, I'm guessing you'll need different things for using hierarchical databases.
